File: find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        
        for(int i=0;i<nums.length;i++)
        {
            if(nums[Math.abs(nums[i])-1]>0)
                nums[Math.abs(nums[i])-1]*=-1;
            
        }
        
        List<Integer> list=new ArrayList<Integer>();
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i]>0)
                list.add(i+1);
        }
        return list;
    }
}

File: find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        
        dict1 = {}
        for i in range(0,len(nums)):
            val=0
            if nums[i] in dict1.keys():
                val =  dict1[nums[i]]
            dict1[nums[i]] = val+1
            
           
        
        res = []
        
        for i in range(1,1+len(nums)):
            if i in dict1.keys():
                continue
            else :
                res.append(i)
                
        return res